### PR TITLE
Update Timer implementation to use std::mutex

### PR DESCRIPTION
**Pull request by pepone**
This PR updates Timer implementation to use `std::mutex` and `std::condition_variable`. Removes the last use of `IceUtil::Mutex` and `IceUtil::Monitor` from the souces.

### CODE REVIEW

File: cpp/src/Ice/Timer.cpp [hunk 1]
```diff
@@ -196,26 +196,7 @@ Timer::run()
...
-                            _monitor.timedWait(timeout);
-                            break;
-                        }
-                        catch(const IceUtil::InvalidTimeoutException&)
-                        {
-                            timeout = timeout / 2;
-                        }
-                    }
-                }
+                _conditionVariable.wait_for(lock, chrono::microseconds((first.scheduledTime - now).toMicroSeconds()));
```
Comment (pepone) [inline cpp/src/Ice/Timer.cpp:199]: This was added to fix https://forums.zeroc.com/discussion/5291/exceptions-can-escape-from-iceutil-timer-run

**Comment by pepone** [comment]
>  Removes the last use of ...

There is actually one more in the metrics code, which I'm removing in a follow up PR.

**Review by externl — APPROVED** [review]

File: cpp/src/Ice/Timer.cpp (hunk 1 above)
Comment (bernardnormier) [reply cpp/src/Ice/Timer.cpp:199]: Here, we want to wait until the scheduledTime. I don't understand why we use wait_for and not wait_until.

Comment (pepone) [reply cpp/src/Ice/Timer.cpp:199]: How do we do the conversion `IceUtil::Time` to `chorono::time_point`? I was planing to replace IceUtil::Time in a follow up PR

**Pull request merged**